django/conf/service_url.py

```diff
@@ -0,0 +1,151 @@
+import inspect
+from collections import defaultdict
+from urllib import parse
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import module_loading
+
+BACKENDS = defaultdict(dict)
+
+
+def parse_url(url):
+    """
+    A method to parse URLs into components that handles quirks
+    with the stdlib urlparse, such as lower-cased hostnames.
+    Also parses querystrings into typed components.
+    """
+    # This method may be called with an already parsed URL
+    if isinstance(url, dict):
+        return url
+    parsed = parse.urlparse(url)
```

Comment:
You can use `urlsplit` instead, it’s a bit faster as it avoids unnecessary parsing

https://youtu.be/ABJvdsIANds